### PR TITLE
roles/cadvisor: Added.

### PR DESCRIPTION
**Pull request by jpds**

### CODE REVIEW

File: ansible/roles/cadvisor/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+####################
+# Docker
+####################
+cadvisor_namespace: "cadvisor"
+cadvisor_image: "{{ docker_registry ~ '/' if docker_registry | default else '' }}{{ cadvisor_namespace }}/cadvisor"
+cadvisor_tag: "v0.28.3"
+cadvisor_image_full: "{{ cadvisor_image }}:{{ cadvisor_tag }}"
+
+cadvisor_restart_policy: "unless-stopped"
+#nodeexporter_restart_retries: 10
```
Comment (markgoddard) [inline ansible/roles/cadvisor/defaults/main.yml]: nit: nodeexporter

File: ansible/roles/cadvisor/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Service deployment definition.
+cadvisor_services:
+  cadvisor:
+    container_name: cadvisor
+    enabled: "{{ cadvisor_enabled }}"
+    image: "{{ cadvisor_image_full }}"
+    published_ports: '8080:8080'
+    read_only: True
+    volumes:
+      - "/:/rootfs"
```
Comment (markgoddard) [inline ansible/roles/cadvisor/defaults/main.yml:16]: Is this required? I don't see anywhere where we're using it.

File: ansible/roles/cadvisor/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      - "/var/run:/var/run:rw"
+      - "/sys:/sys:ro"
+      - "/var/lib/docker/:/var/lib/docker:ro"
```
Comment (markgoddard) [inline ansible/roles/cadvisor/defaults/main.yml:19]: Why is this required?

File: ansible/roles/cadvisor/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+# Whether cAdvisor is enabled.
+cadvisor_enabled: false
+
+# Service deployment definition.
+cadvisor_services:
+  cadvisor:
+    container_name: cadvisor
+    enabled: "{{ cadvisor_enabled }}"
+    image: "{{ cadvisor_image_full }}"
+    published_ports: '8080:8080'
```
Comment (markgoddard) [inline ansible/roles/cadvisor/defaults/main.yml:13]: This is the default port for the inspection store. https://github.com/stackhpc/kayobe/blob/d982be21913cd937eb1a8ae9e58ff944411f3b6f/ansible/group_vars/all/inspector#L329

**Review by markgoddard — CHANGES_REQUESTED** [review]
Thanks. Needs group_vars/all/cadvisor and etc/kayobe/cadvisor.yml.

File: ansible/roles/cadvisor/defaults/main.yml (hunk 3 above)
Comment (jpds) [reply ansible/roles/cadvisor/defaults/main.yml:19]: For the metrics on filesystem usage by containers.

File: ansible/roles/cadvisor/defaults/main.yml (hunk 2 above)
Comment (jpds) [reply ansible/roles/cadvisor/defaults/main.yml:16]: I took this from https://github.com/google/cadvisor - I imagine it's for the machine-wide metrics

**Review by markgoddard — APPROVED** [review]

File: ansible/roles/cadvisor/defaults/main.yml (hunk 4 above)
Comment (markgoddard) [reply ansible/roles/cadvisor/defaults/main.yml:13]: Hmm,  I forgot about this issue, and it hasn't been addressed.

**Pull request merged**